src/pages/integrations/gift-cards/confirm-invoice/confirm-invoice.ts

```diff
@@ -0,0 +1,385 @@
+    this.email = this.merchantProvidedEmail
+      ? this.merchantProvidedEmail
+      : this.buyerProvidedEmail
+      ? this.buyerProvidedEmail
+      : await this.getEmail();
+  }
+
+  ionViewDidLoad() {
+    this.logger.info('Loaded: ConfirmInvoicePage');
+  }
+
+  ionViewWillEnter() {
+    this.isOpenSelector = false;
+    this.navCtrl.swipeBackEnabled = false;
+    const { BITPAY_API_URL } = this.invoiceProvider.credentials;
+    this.network = this.invoiceProvider.getNetwork();
+    this.wallets = this.profileProvider.getWallets({
+      onlyComplete: true,
+      network: this.network,
+      hasFunds: true
+    });
+    this.invoiceUrl = `${BITPAY_API_URL}/invoice/${this.invoiceId}`;
+    const { selectedTransactionCurrency } = this.invoiceData.buyerProvidedInfo;
+    if (selectedTransactionCurrency) {
+      this.wallets = _.filter(this.wallets, (x: any) => {
+        return (
+          x.credentials.coin == selectedTransactionCurrency.toLowerCase() &&
+          !this.profileProvider.vaultHasWallet(x.credentials.walletId)
+        );
+      });
+    }
+    this.wallet = this.wallets[0];
+    if (_.isEmpty(this.wallets)) {
+      this.openInBrowser();
+      return;
+    }
+    if (this.isCordova) {
+      window.addEventListener('keyboardWillShow', () => {
+        this.hideSlideButton = true;
+      });
+
+      window.addEventListener('keyboardWillHide', () => {
+        this.hideSlideButton = false;
+      });
+    }
+    this.paymentTimeControl(this.invoiceData.expirationTime);
+    this.onWalletSelect(this.wallets[0]);
+  }
+
+  public openInBrowser() {
+    this.incomingDataProvider.showMenu({
+      data: this.invoiceUrl,
+      type: 'InvoiceUrl'
+    });
+  }
+
+  private async getEmail() {
+    const email = await this.invoiceProvider.getUserEmail();
+    if (email) {
+      return Promise.resolve(email);
+    } else {
+      return Promise.resolve('');
+    }
+  }
+
+  public async initialize(wallet) {
+    const COIN = wallet.coin.toUpperCase();
+    this.parsedAmount = this.txFormatProvider.parseAmount(
+      wallet.coin,
+      this.invoiceData.price,
+      this.invoiceData.currency
+    );
+    this.currencyIsoCode = this.parsedAmount.currency;
+    this.amountUnitStr = this.parsedAmount.amountUnitStr;
+
+    if (!this.isCryptoCurrencySupported(wallet, this.invoiceData)) {
+      this.onGoingProcessProvider.clear();
+      let msg = this.translate.instant(
+        'Purchases with this cryptocurrency are not enabled'
+      );
+      this.showErrorInfoSheet(msg, null, true);
+      return;
+    }
+    this.onGoingProcessProvider.set('loadingTxInfo');
+
+    // Sometimes API does not return this element;
+    this.invoiceData['minerFees'][COIN]['totalFee'] =
+      this.invoiceData.minerFees[COIN].totalFee || 0;
+    this.invoiceFeeSat = this.invoiceData.minerFees[COIN].totalFee;
+
+    this.message = this.replaceParametersProvider.replace(
+      this.translate.instant(`{{amountUnitStr}} ${this.invoiceName}`),
```

Comment:
What should the invoice message be? 

The GiftCard Invoice message was:
```{{amountUnitStr}} Gift Card```

Reply:
Updated message to say ex: "Payment Request for BitPay Invoice 1234567234 for $1.00 USD to merchant Newegg.com"